pkg/virt-controller/watch/export/export.go

```diff
@@ -384,19 +452,142 @@ func (ctrl *VMExportController) handleIsSourcePvc(vmExport *exportv1.VirtualMach
 			}
 		}
 	} else {
-		if pod.Status.Phase == corev1.PodSucceeded || pod.Status.Phase == corev1.PodFailed {
-			// The server died or completed, delete the pod.
-			ctrl.Recorder.Eventf(vmExport, corev1.EventTypeWarning, exporterPodFailedOrCompletedEvent, "Exporter pod %s/%s succeeded or failed", pod.Namespace, pod.Name)
-			if err := ctrl.Client.CoreV1().Pods(vmExport.Namespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{}); err != nil {
+		if err := ctrl.handlePodSucceededOrFailed(vmExport, pod); err != nil {
+			return 0, err
+		}
+	}
+
+	return ctrl.updateVMExportPvcStatus(vmExport, pvcs, pod, service, inUse)
+
+}
+
+func (ctrl *VMExportController) handleIsVmSnapshot(vmExport *exportv1.VirtualMachineExport, service *corev1.Service) (time.Duration, error) {
+	var pvcs []*corev1.PersistentVolumeClaim
+	vmSnapshot, exists, err := ctrl.getVmSnapshot(vmExport.Namespace, vmExport.Spec.Source.Name)
+	restoreableSnapshots := 0
+	if err != nil {
+		return 0, err
+	} else if exists {
+		if vmSnapshot.Status.ReadyToUse != nil && *vmSnapshot.Status.ReadyToUse {
+			pvcs, restoreableSnapshots, err = ctrl.handlePVCsForVirtualMachineSnapshot(vmExport, vmSnapshot)
+			if err != nil {
+				return 0, err
+			}
+		}
+	} // If snapshot doesn't exist or is not ready, it will fall through to updateVMExportPvcStatus because no pod will exist and no pvcs
+
+	pod, exists, err := ctrl.getExporterPod(vmExport)
+	if err != nil {
+		return 0, err
+	} else if !exists {
+		if len(pvcs) == restoreableSnapshots && restoreableSnapshots > 0 {
```

Comment:
A snapshot may have 0 VolumeSnapshots associated with it.  As-is, the export will be `Pending` forever.  That is probably not the best way to deal with it.  And I'm not exactly sure what is.  More informative conditions are probably a good start.  Maybe even a new phase?

Reply:
So the phase will be `pending`, the ready condition will be `initializing` and the pvc condition will be `pvc not found` in that particular case right now. I could add a snapshot condition in case the source is a snapshot with some information about the snapshots.